racket/collects/racket/contract/private/orc.rkt

```diff
@@ -13,66 +14,76 @@
          (rename-out [_flat-rec-contract flat-rec-contract]
                      [_flat-murec-contract flat-murec-contract]))
 
+(define or/c:none/c (let ([none/c (make-none/c '(or/c))]) none/c))
 (define/subexpression-pos-prop/name or/c-name or/c
-  (case-lambda 
-    [() (make-none/c '(or/c))]
-    [(x) (coerce-contract 'or/c x)]
-    [raw-args
-     (define args (coerce-contracts 'or/c raw-args))
-     (define-values (ho-contracts flat-contracts)
-       (let loop ([ho-contracts '()]
-                  [flat-contracts '()]
-                  [args args])
-         (cond
-           [(null? args) (values (reverse ho-contracts) (reverse flat-contracts))]
-           [else 
-            (let ([arg (car args)])
-              (cond
-                [(flat-contract? arg)
-                 (loop ho-contracts (cons arg flat-contracts) (cdr args))]
-                [else
-                 (loop (cons arg ho-contracts) flat-contracts (cdr args))]))])))
-     (define pred (make-flat-predicate flat-contracts))
-     (define the-or/c
-       (cond
-         [(null? ho-contracts)
+  (case-lambda
+    [() or/c:none/c]
+    [raw-arg*
+     (define raw-args (remq* (list none/c or/c:none/c (first-or/c)) raw-arg*))
```

Comment:
Why not use a `none/c?` predicate here?

Reply:
Because `(or/c ...)` and `(opt/c (or/c ...))` should have the same name.
https://github.com/racket/racket/pull/4437#issuecomment-1255747763

Reply:
I'm not following the answer. Definitely it should be the case that `opt/c` is less important than the regular use, so we should improve `opt/c`. Can you say more about what's going on here?

Reply:
The `contract-name` of `(opt/c (or/c ...))` is checked in the test file [name.rkt](https://github.com/NoahStoryM/racket/blob/andor/pkgs/racket-test/tests/racket/contract/name.rkt#L11).

So if we use `none/c?`, this test will fail:
```racket
> (contract-name (or/c (make-none/c 'none) #t #f))
'boolean?
> (contract-name (opt/c (or/c (make-none/c 'none) #t #f)))
'(or/c none #t #f)
> (test-name 'boolean? (or/c (make-none/c 'none) #t #f))
FAILED (#<procedure:contract-name> #<opt-chaperone-contract: (or/c none #t #f)>)
       got (or/c none #t #f)
  expected boolean?
```

Reply:
Here is the definition of the name of [(opt/c (or/c ...))](https://github.com/NoahStoryM/racket/blob/0be5a4c8938c3cc5b0da6e55bb209f58dcdc3538/racket/collects/racket/contract/private/opters.rkt#L147) .

Reply:
The two options are to adjust the test case to avoid checking `opt/c` or to improve `opt/c` to recognize this case and drop the call to `make-none/c`. I'm happy either way. I think the code that was there, dynamically doing the removal, isn't the best choice, so probably better not to make that more complicated. (If it has to get more complicated, it should be moved into a helper function and a call to the helper function should be generated.)

Reply:
I haven't figured out how to improve `opt/c`, so I just avoid checking it.